lms/templates/dashboard/_dashboard_credit_info.html

```diff
@@ -26,7 +26,7 @@
         </p>
         <div class="purchase_credit">
             ## TODO: set the URL for the link to the provider selection page on the E-Commerce service
-            <a class="btn credit-btn purchase-credit-btn" href="#" target="_blank" data-course-key="${credit_status['course_key']}">${_("Purchase Course Credit")}</a>
+            <a class="btn credit-btn purchase-credit-btn" href="${settings.ECOMMERCE_PUBLIC_URL_ROOT}/credit/checkout/${credit_status['course_key']}" target="_blank" data-course-key="${credit_status['course_key']}">${_("Purchase Course Credit")}</a>
```

Comment:
Use urlparse.urljoin for joining. And make the the complete url in view and send it to template.